src/main/java/uk/gov/hmcts/ccd/domain/service/casedeletion/CaseDataExtractor.java

```diff
@@ -0,0 +1,187 @@
+package uk.gov.hmcts.ccd.domain.service.casedeletion;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import lombok.extern.slf4j.Slf4j;
+import uk.gov.hmcts.ccd.config.JacksonUtils;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseFieldDefinition;
+import uk.gov.hmcts.ccd.domain.model.definition.FieldTypeDefinition;
+import uk.gov.hmcts.ccd.domain.types.BaseType;
+import uk.gov.hmcts.ccd.domain.types.CollectionValidator;
+import uk.gov.hmcts.ccd.domain.types.DocumentValidator;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.inject.Named;
+
+import static uk.gov.hmcts.ccd.domain.model.definition.FieldTypeDefinition.COLLECTION;
+import static uk.gov.hmcts.ccd.domain.model.definition.FieldTypeDefinition.COMPLEX;
+
+@Named
+@Slf4j
+public class CaseDataExtractor {
+    private static final String EMPTY_STRING = "";
+    private static final String FIELD_SEPARATOR = ".";
+
+    public List<String> extractFieldTypePaths(final Map<String, JsonNode> data,
+                                              final List<CaseFieldDefinition> caseFieldDefinitions,
+                                              final String fieldType) {
+        return extractFieldTypePaths(
+            data,
+            caseFieldDefinitions,
+            CaseDataExtractor.EMPTY_STRING,
+            new ArrayList<>(),
```

Comment:
Can this return an immutable list? - e.g. `emptyList()`

Reply:
Done [c129a9a](https://github.com/hmcts/ccd-data-store-api/pull/1872/commits/c129a9a6a6fdff4226e65e0b027691d3cd49352f)